### PR TITLE
contrib: remove install_db4.sh

### PR DESCRIPTION
**Pull request by fanquake**
Now that we can build a bdb-only depends prefix (#26833), there is no need to
maintain a bdb-building bash script, that does the same thing as
depends, except worse, as it's missing patches and workarounds. i.e #26623.

Someone that wants to compile bdb themselves, but doesn't want to use other depends built libs, can do:
```bash
make -C depends NO_BOOST=1 NO_LIBEVENT=1 NO_QT=1 NO_SQLITE=1 NO_NATPMP=1 NO_UPNP=1 NO_ZMQ=1 NO_USDT=1
...
to: /path/to/bitcoin/depends/x86_64-pc-linux-gnu
```

which gives them a BDB only prefix, and then compile using:
```bash
export BDB_PREFIX="/path/to/bitcoin/depends/x86_64-pc-linux-gnu"
./autogen.sh
./configure \
    BDB_LIBS="-L${BDB_PREFIX}/lib -ldb_cxx-4.8" \
    BDB_CFLAGS="-I${BDB_PREFIX}/include"
```

Wondering if we should extract the build bdb/legacy wallet docs somewhere, to avoid the repetition?

### CODE REVIEW

**Comment by murrayn** [comment]
Concept ACK

File: doc/build-openbsd.md [hunk 1]
```diff
@@ -41,16 +41,18 @@ pkg_add sqlite3
 BerkeleyDB is only required to support legacy wallets.
 
 It is recommended to use Berkeley DB 4.8. You cannot use the BerkeleyDB library
-from ports. However you can build it yourself, [using the installation script included in contrib/](/contrib/install_db4.sh), like so, from the root of the repository.
+from ports. However you can build it yourself, [using depends](/depends).
 
 ```bash
-./contrib/install_db4.sh `pwd`
+make -C depends NO_BOOST=1 NO_LIBEVENT=1 NO_QT=1 NO_SQLITE=1 NO_NATPMP=1 NO_UPNP=1 NO_ZMQ=1 NO_USDT=1
```
Comment (murrayn) [inline doc/build-openbsd.md:47]: This should be `gmake` for OpenBSD.

Comment (fanquake) [reply doc/build-openbsd.md]: Thanks. Fixed up.

File: doc/build-unix.md [hunk 2]
```diff
@@ -197,17 +197,21 @@ Berkeley DB
 -----------
 
 The legacy wallet uses Berkeley DB. To ensure backwards compatibility it is
-recommended to use Berkeley DB 4.8. If you have to build it yourself, you can
-use [the installation script included in contrib/](/contrib/install_db4.sh)
-like so:
-
-```shell
-./contrib/install_db4.sh `pwd`
+recommended to use Berkeley DB 4.8. If you have to build it yourself, and don't
```
Comment (murrayn) [inline doc/build-unix.md:200]: Wondering if this section should be titled `Legacy Wallet Support` rather than `Berkeley DB`?

**Comment by hebasto** [comment]
Concept ACK.

File: doc/build-unix.md [hunk 3]
```diff
@@ -197,17 +197,21 @@ Berkeley DB
...
+```bash
+export BDB_PREFIX="/path/to/bitcoin/depends/x86_64-pc-linux-gnu"
 
-from the root of the repository.
-
-Otherwise, you can build Bitcoin Core from self-compiled [depends](/depends/README.md).
+./configure \
+    BDB_LIBS="-L${BDB_PREFIX}/lib -ldb_cxx-4.8" \
+    BDB_CFLAGS="-I${BDB_PREFIX}/include"
+```
```
Comment (hebasto) [inline doc/build-unix.md:214]: Actually, for an intention like _"don't want to use any other libraries built in depends"_ there is a dedicated `ALLOW_HOST_PACKAGES=1` dependencies option:

```bash
$ make -C depends NO_BOOST=1 NO_LIBEVENT=1 NO_QT=1 NO_SQLITE=1 NO_NATPMP=1 NO_UPNP=1 NO_ZMQ=1 NO_USDT=1 ALLOW_HOST_PACKAGES=1
$ ./configure CONFIG_SITE=$PWD/depends/x86_64-pc-linux-gnu/share/config.site --with-gui --with-sqlite --enable-zmq --with-miniupnpc --with-natpmp --enable-usdt
```

Comment (fanquake) [reply doc/build-unix.md:214]: > ALLOW_HOST_PACKAGES=1 dependencies option:

I don't particularly like the existence of that option in general, or want to introduce it here. So given that we acheive the same thing here (only bdb in the prefix) regardless of whether it's used, I'd rather not add it.

Comment (fanquake) [reply doc/build-unix.md:214]: That is also a somwhat more convoluted/confusing `./configure` invocation, as opposed to just passing the flags.

Comment (hebasto) [reply doc/build-unix.md:214]: > That is also a somewhat more convoluted/confusing `./configure` invocation

True.

> as opposed to just passing the flags.

... which is error-prone: if a typo slipped in `BDB_LIBS` or `BDB_CFLAGS`, the `configure` script is not testing them and only following `make` will fail.

**Review by hebasto — APPROVED** [review]
ACK 44f3c7de21fd86343a2427d2e864ea2001499c5e

File: doc/build-unix.md (hunk 3 above)
Comment (fanquake) [reply doc/build-unix.md:214]: > which is error-prone:

That's hardly any worse than making users type out all this `./configure CONFIG_SITE=$PWD/depends/x86_64-pc-linux-gnu/share/config.site --with-gui --with-sqlite --enable-zmq --with-miniupnpc --with-natpmp --enable-usdt`, where any typo or forgotten flag would result in missing features that wouldn't be noticed until runtime.

Comment (hebasto) [reply doc/build-unix.md:214]: Users can read "Options used to compile and link:" output of the `configure` script to ensure all requested features are indeed enabled. But we cannot rely on users' reasonable behavior :)

Comment (hebasto) [reply doc/build-unix.md:214]: I agree with either approach, so labeling this comment as resolved.

**Review by jarolrod — APPROVED** [review]
ACK 44f3c7de21fd86343a2427d2e864ea2001499c5e

Did not test openbsd build instructions, but they are appropriate for this PR as that build doc instructed to use the db script.

**Comment by TheCharlatan** [comment]
ACK 44f3c7de21fd86343a2427d2e864ea2001499c5e

Disabling the other targets to enable building only a specific build target seems a bit odd though. How about adding a `$(package)_install` target instead? Then we could call `make bdb_install` and get the same result, but with saner arguments. I made a small proof of concept here (just for building and installing, does not generate the config.site yet): https://github.com/TheCharlatan/bitcoin/pull/6 . Would this be potentially dangerous, because it skips over other installation steps that are required e.g. for otool? If so, I'd rather stick to the solution used in this PR than maintaining yet another way to use depends.

**Review by sedited — APPROVED** [review]

**Comment by achow101** [comment]
ACK 44f3c7de21fd86343a2427d2e864ea2001499c5e

**Comment by Sjors** [comment]
Post merge: just noticed the script was gone, but the bdb-only-depends approach works.

**Pull request merged**